graph-builder/src/registry.rs

```diff
@@ -394,3 +496,56 @@ fn assemble_metadata(blob: &[u8], metadata_filename: &str) -> Result<Metadata, E
     }
     .map_err(Into::into)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn registry_try_parse_valid() {
+        let tests = vec![
```

Comment:
Can you please add an IPv4 test for `127.0.0.1`? The regexps should be already matching that.